src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs

```diff
@@ -1154,6 +1154,492 @@ public static Vector<T> Negate<T>(Vector<T> value)
         public static Vector<T> OnesComplement<T>(Vector<T> value)
             where T : struct => ~value;
 
+        /// <summary>Shifts each element of a vector left by the specified amount.</summary>
+        /// <param name="value">The vector whose elements are to be shifted.</param>
+        /// <param name="shiftCount">The number of bits by which to shift each element.</param>
+        /// <returns>A vector whose elements where shifted left by <paramref name="shiftCount" />.</returns>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector<byte> ShiftLeft(Vector<byte> value, int shiftCount)
+        {
+            Unsafe.SkipInit(out Vector<byte> result);
+
+            for (int index = 0; index < Vector<byte>.Count; index++)
+            {
+                var element = Scalar<byte>.ShiftLeft(value.GetElementUnsafe(index), shiftCount);
```

Comment:
Nit: can we avoid using var here and elsewhere?  This is a good example of where I don't actually know what type element is.

Reply:
I can do a separate PR to fix it up for all the vector code. It has (and has for a long time) used `var` fairly extensively to help make copying the algorithms around easier.

The "type" really doesn't matter, the action being done does and that's generally guaranteed by the generic context (e.g. `Scalar<byte>` means its operating on `byte`).

Reply:
> I can do a separate PR to fix it up for all the vector code

Thanks